tests/integration/StorageWriter/IPFS.test.ts

```diff
@@ -0,0 +1,97 @@
+import { describe } from 'riteway'
+import fetch from 'node-fetch'
+
+import { IPFS } from '../../../src/StorageWriter/IPFS'
+import { normalClaim, invalidUtf8CharactersClaim, encodeContent, decodeContent, cleanContent } from './claimData'
+
+const IPFS_URL = process.env.IPFS_URL || 'http://localhost:5001'
+
+const createIPFS = ({
+  ipfsUrl = IPFS_URL
+} = {}) => {
+  return new IPFS({
+    ipfsUrl
+  })
+}
+
+const fetchFile = async (hash: string): Promise<string> => {
+  const response = await fetch(`${IPFS_URL}/api/v0/cat?arg=${hash}`)
+  return response.text()
+}
+
+describe('IPFS.addText', async should => {
+  const { assert } = should('match the claim read from ipfs')
+
+  {
+    const ipfs = createIPFS()
+    const claim = normalClaim
+    let hash, claimFromIPFS
+    
+    try {
+      hash = await ipfs.addText(JSON.stringify(claim))
```

Comment:
hash should be a const, and not defined above as let above. Same with the rest of the tests.